code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -113,43 +113,71 @@
 
 /obj/item/gun/energy/lwap
 	name = "LWAP laser sniper"
-	desc = "A highly advanced laser sniper that does more damage the farther away the target is, but fires slowly."
+	desc = "A highly advanced laser sniper that does more damage the farther away the target is, but fires slowly. Comes with a super advanced scope, which can highlight threats through walls, and pierce one object, after being deployed for a while."
 	icon_state = "esniper"
 	item_state = null
 	w_class = WEIGHT_CLASS_BULKY
 	force = 12
-	flags =  CONDUCT
+	flags = CONDUCT
 	slot_flags = SLOT_BACK
 	can_holster = FALSE
 	weapon_weight = WEAPON_HEAVY
 	origin_tech = "combat=6;magnets=6;powerstorage=4"
-	ammo_type = list(/obj/item/ammo_casing/energy/laser/sniper)
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/sniper, /obj/item/ammo_casing/energy/laser/sniper/pierce)
 	zoomable = TRUE
 	zoom_amt = 7
 	shaded_charge = TRUE
+	var/scope_active = FALSE //To ensure we don't activate the piercing ammo out of scope.
+
+/obj/item/gun/energy/lwap/zoom(mob/living/user, forced_zoom)
+	. = ..()
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(scope_active)
+			select_fire(H)
+			H.remove_status_effect(STATUS_EFFECT_LWAPSCOPE)
+			scope_active = FALSE
+		else if(do_after(user, 3 SECONDS, target = src))
+			if(zoomed) //We check after.
+				scope_active = TRUE
+				to_chat(user, "<b><span class='robot'>SMRT-SCOPE Online.</span></b>")
```

Comment:
Just to be sure, its supposed to be SMRT or SMART?

Reply:
whatever I decide to name it, SMRT seemed funny, but also have been asking around for ironic / good names for a "smart" scope

Reply:
QCK-SCOPE

Reply:
actually nevermind SCOPE-CREEP is infinitely funnier and fits the ability more